metronome/checkpointing/models/src/io/iohk/metronome/checkpointing/CheckpointingAgreement.scala

```diff
@@ -1,26 +1,29 @@
 package io.iohk.metronome.checkpointing
 
-import io.iohk.ethereum.crypto.ECDSASignature
 import io.iohk.metronome.crypto
 import io.iohk.metronome.hotstuff.consensus.ViewNumber
-import io.iohk.metronome.hotstuff.consensus.basic.{Agreement, VotingPhase}
-import org.bouncycastle.crypto.params.{
-  ECPublicKeyParameters,
-  ECPrivateKeyParameters
+import io.iohk.metronome.hotstuff.consensus.basic.{
+  Secp256k1Agreement,
+  VotingPhase
 }
+import scodec.bits.ByteVector
+import io.iohk.ethereum.rlp
+import io.iohk.metronome.checkpointing.models.RLPCodecs._
 
-object CheckpointingAgreement extends Agreement {
+object CheckpointingAgreement extends Secp256k1Agreement {
   override type Block = models.Block
   override type Hash  = models.Block.Header.Hash
-  override type PSig  = ECDSASignature
-  // TODO (PM-2935): Replace list with theshold signatures.
-  override type GSig = List[ECDSASignature]
-  override type PKey = ECPublicKeyParameters
-  override type SKey = ECPrivateKeyParameters
 
   type GroupSignature = crypto.GroupSignature[
     PKey,
     (VotingPhase, ViewNumber, Hash),
     GSig
   ]
+
+  def signedContentSerialiser(
+      phase: VotingPhase,
+      viewNumber: ViewNumber,
+      hash: Hash
+  ): ByteVector =
+    ByteVector(rlp.encode(phase) ++ rlp.encode(viewNumber) ++ rlp.encode(hash))
```

Comment:
I wonder if the compiler would find the type class later it if this was like:

```
object CheckpointingAgreement extends Secp256k1Agreement {
  ... 

  implicit val signing: Signing[CheckpointingAgreement] = 
    Signing.secp256k1((phase, viewNumber, hash) => 
      ByteVector(rlp.encode(phase) ++ rlp.encode(viewNumber) ++ rlp.encode(hash))
    )
}
```

It seemed to have [worked for Block](https://github.com/input-output-hk/metronome/blob/3fe4bf72afb3b8e3bce1eff52ebfc076e9811b58/metronome/checkpointing/models/src/io/iohk/metronome/checkpointing/CheckpointingAgreement.scala#L28).